compiler/src/dmd/target.d

```diff
@@ -1136,6 +1136,15 @@ extern (C++) struct Target
     {
         return (os & Target.OS.Posix) != 0;
     }
+
+    /*********************
+     * Returns:
+     *  alignment of the stack
+     */
+    extern (C++) uint stackAlign()
```

Comment:
Maximum? Minimum? Preferred? Main function or non-main function? This interface is not very clear.

Reply:
It's the stack alignment the compiler happens to be generating for the target.

Reply:
That is really dependent on whether the OS expects a specific stack boundary alignment (ie: main function), what minimum alignment must the target apply (compat with extern(C/C++), or what types does the function handle on stack (ie the alignment set is really _dynamic_, and we just have a lower and upper bound to constrain it).

Reply:
> You're saying that gdc and ldc don't need it. That's ok. DMD does. My experiments with gdc and ldc show that this is the default stack alignment they use.

`extern(C) main` function has a higher stack alignment requirement though.  What I can see from a quick grep.
- Darwin is always 16 bytes
- All other x86_64's are 16 bytes.
- All other x86's are 4 bytes.

Reply:
> My experiments with gdc and ldc show that this is the default stack alignment they use.

Actually, the default stack alignment used by all x86/64's is 16 bytes, to support SSE whether the x86 target has it or not.

Reply:
My experiments showed otherwise. Perhaps it's just a different version of the compiler I tried. In any case, it simply does not matter here. This is the alignment dmd uses, and it works.